v2/router.go

```diff
@@ -40,14 +40,42 @@ type TransportProcedure struct {
 	// HandlerSpec specifying which handler and rpc type.
 	HandlerSpec TransportHandlerSpec
 
-	// Encoding of the handler.
-	// (if present).
+	// Encoding of the handler (if present).
 	Encoding Encoding
 
 	// Human-readable signature of the handler.
 	Signature string
 }
 
+// EncodingProcedure specifies a single encoding-level handler. The RouteTable can register an
+// EncodingProcedure after mapping it to a corresponding TransportProcedure.
+type EncodingProcedure struct {
```

Comment:
can we add a `MarshalObject` implementation similar to the  `TransportProcedure` below? This plays nicely with zap logging